### PR TITLE
Userspice PostgreSQL Installer

### LINKED ISSUE #5: Fix installer
Please fix the installer files to work with PostgreSQL in a fully compatible way:

- Pre-installation checks should work
- Connection testing should work (not necessarily with the current MySQL way of creating a new DB + user, if there is a simpler way)
- Automatic DB creation should work on missing DB, if possible in PostgreSQL
- Automatic execution of `sql.sql` should work (not necessarily with the current MySQL way of executing it line by line, IF PostgreSQL is capable of executing a full SQL file and returning the problematic line on error)

### PR DESCRIPTION
**Pull request by ssobii2**
Please check and let me know if anything needs changing.

I checked these use cases:
1. If there is no db already created, it makes it for you with no issues assuming you are using the default Postgres superuser.
2. If the db is already created, it still works, assuming you are using the default Postgres superuser.
3. If the db is not already created or is already there in both cases, assuming you don't want to use the default Postgres superuser, then you need to create a new user with the necessary privileges; first, it doesn't create it for you.
4. If a new user is created, it works the same way as using the default superuser with no issues.

When it first time redirects you to the login page, there is an error idk why. Screenshot of the error below, but once you refresh the page, the error is gone, and you can log in normally.

image

### CODE REVIEW

File: install/index.php
Comment (GaborFarkas) [inline install/index.php:1]: Please do the same in the other index.php in the install folder. I don't know why it's there as it's seemingly not used anywhere, but we should keep our development consistent.

File: users/updates/components/2020-04-17a.php
Comment (GaborFarkas) [inline users/updates/components/2020-04-17a.php:1]: `ADD` is the correct one here? Not `ALTER`? Are the columns already there or are they new at this point?

**Review by GaborFarkas — CHANGES_REQUESTED** [review]

**Comment by GaborFarkas** [comment]
Most importantly, these 3 commits are also part of a previous PR. Please make sure to remove them from this one.

image

File: install/index.php
Comment (ssobii2) [reply install/index.php:1]: Ok will do that professor

File: users/updates/components/2020-04-17a.php
Comment (ssobii2) [reply users/updates/components/2020-04-17a.php:1]: I will check it, if no then i will change to `ALTER`

**Comment by ssobii2** [comment]
> Most importantly, these 3 commits are also part of a previous PR. Please make sure to remove them from this one.
> 
> image

I used rebase, but in the sql.sql file, some changes were not the latest, so I added them again rest is fine with the new commit.

**Review by GaborFarkas — APPROVED** [review]

**Comment by GaborFarkas** [comment]
Please rebase this branch on the master branch to resolve conflicts.

**Comment by ssobii2** [comment]
> Please rebase this branch on the master branch to resolve conflicts.

Do I have access to rebase master branch? Also it says resolve conflict here. Can’t we do it from here?

**Comment by GaborFarkas** [comment]
> > Please rebase this branch on the master branch to resolve conflicts.
> 
> Do I have access to rebase master branch? Also it says resolve conflict here. Can’t we do it from here?

You should rebase THIS branch on master. You can read the master branch to do this. Please do it ASAP.

**Comment by ssobii2** [comment]
> > > Please rebase this branch on the master branch to resolve conflicts.
> > 
> > 
> > Do I have access to rebase master branch? Also it says resolve conflict here. Can’t we do it from here?
> 
> You should rebase THIS branch on master. You can read the master branch to do this. Please do it ASAP.

Ok i am doing it now

**Comment by ssobii2** [comment]
> > > Please rebase this branch on the master branch to resolve conflicts.
> > 
> > 
> > Do I have access to rebase master branch? Also it says resolve conflict here. Can’t we do it from here?
> 
> You should rebase THIS branch on master. You can read the master branch to do this. Please do it ASAP.

Hi Professor,
It's not letting me do this. I tried doing it, and I solved this conflict, but then it opened other files conflict idk why. I am late for class. Let me know the solution. I will look into it after class.

image

**Comment by ssobii2** [comment]
> > > > Please rebase this branch on the master branch to resolve conflicts.
> > > 
> > > 
> > > Do I have access to rebase master branch? Also it says resolve conflict here. Can’t we do it from here?
> > 
> > 
> > You should rebase THIS branch on master. You can read the master branch to do this. Please do it ASAP.
> 
> Hi Professor, It's not letting me do this. I tried doing it, and I solved this conflict, but then it opened other files conflict idk why. I am late for class. Let me know the solution. I will look into it after class.
> 
> image

I even tried cloning the repo again, and I did all the settings again, but still, it's asking to fix more and if i fix this it asks for more update files conflicts. As you can see, it just committed 2 extra commits, which were already there, there were part of previous pull request.

Wouldn't it be better to just do it from here? I did it from here. I found this solution here

https://stackoverflow.com/questions/31863429/how-do-i-resolve-conflicts-with-git

**Pull request merged**